numpy/core/src/multiarray/item_selection.c

```diff
@@ -2236,14 +2236,17 @@ PyArray_Nonzero(PyArrayObject *self)
         static npy_intp const zero_dim_shape[1] = {1};
         static npy_intp const zero_dim_strides[1] = {0};
 
+        Py_INCREF(PyArray_DESCR(self));
         PyArrayObject *self_1d = (PyArrayObject *)PyArray_NewFromDescrAndBase(
             Py_TYPE(self), PyArray_DESCR(self),
             1, zero_dim_shape, zero_dim_strides, PyArray_BYTES(self),
             PyArray_FLAGS(self), (PyObject *)self, (PyObject *)self);
         if (self_1d == NULL) {
             return NULL;
         }
-        return PyArray_Nonzero(self_1d);
+        ret_tuple = PyArray_Nonzero(self_1d);
+        Py_DECREF(self_1d);
+        return ret_tuple;
```

Comment:
Oops, this was recent.

Reply:
I could split off the commit to merge quicker and make sure it is in 1.17.

Reply:
Probably not a big deal, this path is unlikely to ever be hit anyway (hence my attempt to deprecate it)